calblueprint/sundayfriends
# Albert/invites form validation

Added form validation for invite admin modal, including email validation. 

Note: @gabe I didn't include anything on the frontend to handle error case where an admin is reinvited after already becoming an admin — there doesn't seem to be anything in the Figma for that (it's handled in the firebase call tho so they won't be re-added in firebase). IMO it would be kind of clunky to redirect users to a new modal to handle the case where an invite has already been sent to a specific email address — can check with Kelly and Cindy on this too.

## Testing

- Ensure correct error state appears when "name" field is empty
- Ensure correct error state appears when "email" field is empty
- Ensure correct error state appears when "email" field has bad email address
- Ensure form does not send while error states are active
- Ensure spacing between text fields remains consistent when form is given bad inputs

_List all manual tests done/new unit tests you have implemented_
Performed manual testing and checked for the five above conditions.

## Screenshots

<img width="1389" alt="Screen Shot 2021-11-30 at 1 41 11 AM" src="https://user-images.githubusercontent.com/25868533/144026170-fa41d682-f7d9-4462-93c1-3f68e3f8ebcc.png">
<img width="1393" alt="Screen Shot 2021-11-30 at 1 43 34 AM" src="https://user-images.githubusercontent.com/25868533/144026196-20cf4099-a0db-41bd-9c75-e24f146c0229.png">

CC: @gabriel-jt @cindyxzhang @hukellyy
